src/ui/toast.js

```diff
@@ -106,21 +110,58 @@ export class Toast {
           return port.whenReady();
         }).then(() => {
           resetStyles(this.iframe_, ['height']);
-          setImportantStyles(this.iframe_, {
-            'animation': 'swg-notify .3s ease-out normal backwards, '
-                  + 'swg-notify-hide .3s ease-out ' + toastDurationSeconds +
-                  's normal forwards',
+
+          this.animate_(() => {
+            setImportantStyles(this.iframe_, {
+              'transform': 'translateY(100%)',
+              'opactiy': 1,
+              'visibility': 'visible',
+            });
+            transition(this.iframe_, {
+              'transform': 'translateY(0)',
+              'opacity': 1,
+              'visiblity': 'visible',
+            }, 400, 'ease-out');
           });
+
+          // Close the Toast after the specified duration.
           this.doc_.getWin().setTimeout(() => {
             this.close();
           }, (toastDurationSeconds + 1) * 1000);
         });
   }
 
+    /**
+   * @param {function():!Promise} callback
+   * @return {!Promise}
+   * @private
+   */
+  animate_(callback) {
+    const wait = this.animating_ || Promise.resolve();
+    return this.animating_ = wait.then(() => {
+      return callback();
+    }, () => {
+      // Ignore errors to make sure animations don't get stuck.
+    }).then(() => {
+      this.animating_ = null;
+    });
+  }
+
   /**
    * Closes the toast.
    */
   close() {
-    this.doc_.getBody().removeChild(this.iframe_);
+    this.animate_(() => {
+      transition(this.iframe_, {
+        'transform': 'translateY(100%)',
+        'opacity': 1,
+        'visiblity': 'visible',
+      }, 400, 'ease-out');
+    });
+
+    // Remove the toast from the DOM after animation is complete.
+    this.doc_.getWin().setTimeout(() => {
```

Comment:
I'd recommend returning a Promise here for ease of testing and/or nesting in code.

Reply:
Done